ezmsg/core/subclient.py

```diff
@@ -121,24 +115,61 @@ async def _disconnect_publisher(self, id: UUID):
                 with suppress(asyncio.CancelledError):
                     await task
 
-    async def _handle_publisher(self, id: UUID, reader: asyncio.StreamReader) -> None:
+    async def _handle_publisher(self, id: UUID, address: Address, connected: asyncio.Event) -> None:
+
+        reader, writer = await asyncio.open_connection(*address)
+        writer.write(encode_str(str(self.id)))
+        writer.write(uint64_to_bytes(self.pid))
+        writer.write(encode_str(self.topic))
+        await writer.drain()
+        pub_id_str = await read_str(reader)
+        pub_pid = await read_int(reader)
+        pub_topic = await read_str(reader)
+        num_buffers = await read_int(reader)
+
+        if id != UUID(pub_id_str):
+            raise ValueError("Unexpected Publisher ID")
+
+        # NOTE: Not thread safe
```

Comment:
Thankfully, no; the current design of ezmsg is not thread safe generally.  acquiring a mutex for every cache access could really slow things down, and we can generally handle parallelism via multiprocessing instead of threading